src/main/java/be/seeseemelk/mockbukkit/block/state/SignMock.java

```diff
@@ -214,4 +198,99 @@ public void setWaxed(boolean waxed)
 		return new SignMock(this);
 	}
 
+	private static class SignSideImpl implements SignSide {
+
+		private final Component[] lines;
+		private boolean glowing = false;
+		private DyeColor color = DyeColor.BLACK;
+
+		private SignSideImpl() {
+			this.lines = new Component[4];
+			Arrays.fill(lines, Component.empty());
+		}
+
+		private SignSideImpl(SignSide signSide) {
+			this.lines = signSide.lines().toArray(new Component[0]);
+			this.glowing = signSide.isGlowingText();
+			this.color = signSide.getColor();
+		}
+
+		@Override
+		public @NotNull List<Component> lines()
+		{
+			return List.of(lines);
+		}
+
+		@Override
+		public @NotNull Component line(int index) throws IndexOutOfBoundsException
+		{
+			if (index < 0 || index >= lines.length)
+				throw new IndexOutOfBoundsException("Index out of bounds: " + index);
+			return lines[index];
+		}
+
+		@Override
+		public void line(int index, @NotNull Component line) throws IndexOutOfBoundsException
+		{
+			Preconditions.checkNotNull(line, "Line cannot be null!");
+			if (index < 0 || index >= lines.length)
+				throw new IndexOutOfBoundsException("Index out of bounds: " + index);
+			lines[index] = line;
+		}
+
+		@Override
+		public @NotNull String[] getLines()
+		{
+			return Arrays.stream(lines)
+					.map(LegacyComponentSerializer.legacySection()::serialize)
+					.toArray(String[]::new);
+		}
+
+		@Override
+		public @NotNull String getLine(int index) throws IndexOutOfBoundsException
+		{
+			if (index < 0 || index >= lines.length)
+				throw new IndexOutOfBoundsException("Index out of bounds: " + index);
+			return LegacyComponentSerializer.legacySection().serialize(lines[index]);
+		}
+
+		// Please note: NullableProblems is suppressed because the method signature requires a non-null String but
+		// the implementation allows null values to be set.
+		@Override
+		public void setLine(int index, @SuppressWarnings("NullableProblems") String line) throws IndexOutOfBoundsException
+		{
+			if (index < 0 || index >= lines.length)
+				throw new IndexOutOfBoundsException("Index out of bounds: " + index);
+			if (line == null) {
+				lines[index] = Component.empty();
+			} else {
+				lines[index] = LegacyComponentSerializer.legacySection().deserialize(line);
+			}
+		}
+
+		@Override
+		public boolean isGlowingText()
+		{
+			return glowing;
+		}
+
+		@Override
+		public void setGlowingText(boolean glowing)
+		{
+			this.glowing = glowing;
+		}
+
+		@Override
+		public @NotNull DyeColor getColor()
+		{
+			return color;
+		}
+
+		@Override
+		public void setColor(DyeColor color)
```

Comment:
Please add a `@NotNull` annotation here

Reply:
Done in 019dd7dbcec9e72bc984dd4b454296567e671707